File: scope_ir.py
from span import SourceSpan


class ScopeIR:
    def __init__(self, id: int, name: str, kind: str, parent_id: int, span: SourceSpan):
        self.id=id,
        self.name=name,
        self.kind=kind,
        self.parent_id=parent_id,
        self.span=span

File: rust/src/ir/scope_ir.rs
use crate::ir::nodes::span_ir::SourceSpan;


#[derive(Debug, Clone)]
pub struct ScopeIR {
    pub id: i64,
    pub name: String,
    pub kind: String, // todo change to custom enum
    pub parent_id: i64,
    pub span: SourceSpan
}